File: benchmarks/tsb/bench_dt_total_seconds.ts
/**
 * Benchmark: DatetimeAccessor.total_seconds — epoch-second conversion on 100k datetime Series.
 * Outputs JSON: {"function": "dt_total_seconds", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { Series } from "../../src/index.ts";

const SIZE = 100_000;
const WARMUP = 5;
const ITERATIONS = 50;

const base = new Date("2020-01-01T00:00:00Z").getTime();
const dates = Array.from({ length: SIZE }, (_, i) => new Date(base + i * 60_000));
const s = new Series({ data: dates });

for (let i = 0; i < WARMUP; i++) {
  s.dt.total_seconds();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.dt.total_seconds();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "dt_total_seconds",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_dt_total_seconds.py
"""Benchmark: Series.dt.total_seconds() — epoch-second conversion on 100k datetime Series."""
import json, time
import pandas as pd
import numpy as np

SIZE = 100_000
WARMUP = 5
ITERATIONS = 50

base = pd.Timestamp("2020-01-01T00:00:00Z")
dates = pd.date_range(start=base, periods=SIZE, freq="min")
s = pd.Series(dates)

for _ in range(WARMUP):
    (s - pd.Timestamp("1970-01-01", tz="UTC")).dt.total_seconds()

times = []
for _ in range(ITERATIONS):
    t0 = time.perf_counter()
    (s - pd.Timestamp("1970-01-01", tz="UTC")).dt.total_seconds()
    times.append((time.perf_counter() - t0) * 1000)

total_ms = sum(times)
print(json.dumps({"function": "dt_total_seconds", "mean_ms": round(total_ms / ITERATIONS, 3), "iterations": ITERATIONS, "total_ms": round(total_ms, 3)}))
